internal/provider/kubernetes/helpers.go

```diff
@@ -197,12 +197,20 @@ func refsSecret(ref *gwapiv1b1.SecretObjectReference) bool {
 		(ref.Kind == nil || *ref.Kind == gatewayapi.KindSecret)
 }
 
-func infraServiceName(gateway *gwapiv1b1.Gateway) string {
+func infraServiceName(gateway *gwapiv1b1.Gateway, merged bool) string {
+	if merged {
```

Comment:
can we simplify this
```
var ns, name string
if  merged {
  ns, name = "", string(gateway.Spec.GatewayClassName)
} else {
  ns, name = gateway.Namespace , gateway.Name
}
 ......
```

Reply:
true, also I have merged `infraServiceName` and `infraDeploymentName` as they were the same